post_condor.yml

```diff
@@ -29,9 +33,41 @@ targets:
   1_get_data_ready/out:
     command: dir.create(target_name, recursive=I(TRUE), showWarnings=I(FALSE))
   
-  master_list:
-    command: readRDS(file = I(gd_get('1_get_data_ready/out/merged_flow.rds.ind')))
+  1_get_data_ready/out/merged_flow.rds.ind:
+    command: gd_get(target_name)
+    
+  1_get_condor_data/out/master_list.rds:
+    command: saveRDS(object= I(readRDS(as_data_file("1_get_data_ready/out/merged_flow.rds.ind"))), file = I("1_get_condor_data/out/master_list.rds"))
 
   1_get_condor_data/out/modeled.rds.ind:
-    command: get_data(master_list, target_name)
+    command: get_data("1_get_condor_data/out/master_list.rds", target_name)
     
+  1_get_condor_data/out/modeled.rds:
+    command: gd_get('1_get_condor_data/out/modeled.rds.ind')
+    
+  1_get_condor_data/out/master_list.rds.ind:
+    command: gd_put(remote_ind=target_name, local_source="1_get_condor_data/out/master_list.rds")
+
+  master_list:
+    command: get_master_list("1_get_condor_data/out/master_list.rds.ind")
+
+  1_get_condor_data/doc:
+    command: dir.create(target_name, recursive=I(TRUE), showWarnings=I(FALSE))
+  
+  1_get_condor_data/doc/plot.pdf.ind:
+    command: plot_data(master_list,"1_get_condor_data/out/modeled.rds", target_name)
```

Comment:
Up to you, but you could make the second argument `ind_file="1_get_condor_data/out/modeled.rds.ind"` instead of `data_file="1_get_condor_data/out/modeled.rds"`, then use `data_file <- sc_retrieve(ind_file)`  within `plot_data()`. This would have the effect of making it so that after you've built and gd_put the whole project, any colleague who calls `scmake('1_get_condor_data/doc/plot.pdf')` will simply pull rather than rebuilding the plot.

Reply:
Sigh...that's what I was going for. I guess I'm not clear why they would re-build the plots. Don't I have the logic correct? The last thing it's looking for is the local plot. It gets it via gd_get, and gets and indicator file then. That means, it doesn't need to rebuild the previous step, correct?

Reply:
Oh...maybe I get it. Because even though the indicator file itself is updated, the argument in there isn't....gottcha

Reply:
Hmm, you're right that the plot itself won't get rebuilt. But there'll be some unnecessary steps along the way. I think it'd go like this:
* Collaborator calls `scmake('1_get_condor_data/doc/plot.pdf')`
* remake sees that 1_get_condor_data/doc/plot.pdf depends on 1_get_condor_data/doc/plot.pdf.ind depends on 1_get_condor_data/out/modeled.rds depends on 1_get_condor_data/out/modeled.rds.ind depends on 1_get_condor_data/out/master_list.rds, and so on. Because not all of these files exist locally, 1_get_condor_data/doc/plot.pdf is initially out of date ('dirty') because (1) it's missing and also (2) one or more of its ancestors is missing (remake calls this 'dirty by descent'). remake starts with the ancestors:
* 1_get_condor_data/out/master_list.rds doesn't exist locally, so remake considers it out of date and rebuilds it. This could have been avoided.
* Now remake can confirm, by looking at file hashes, that 1_get_condor_data/out/modeled.rds.ind is already up to date. Good!
* 1_get_condor_data/out/modeled.rds doesn't exist locally, so remake builds the target (calls gd_get). This also could have been avoided.
* Now remake can confirm by hashes that 1_get_condor_data/doc/plot.pdf.ind is up to date. Good!
* 1_get_condor_data/doc/plot.pdf doesn't exist locally, so remake builds the target (calls gd_get). This is what we wanted.

Reply:
Just saw your comment from 12 minutes ago. Yes, exactly.

Reply:
OK, yeah, I think it's settling in... In my defense and completely in hindsight (:smiley: ) ...I do want my collaborators mainly to have locally: "1_get_condor_data/out/modeled.rds" and "1_get_condor_data/out/master_list.rds". The plots are just fluff....

Reply:
Hmm, that's an intriguing goal I hadn't thought of before - if folks don't mind the downloads, this is kind of a cool way for you to shape exactly which set of files end up on their computers the first time they build.